core/chains/evm/gas/block_history_estimator.go

```diff
@@ -232,7 +232,7 @@ func (b *BlockHistoryEstimator) GetDynamicFee(gasLimit uint64) (fee DynamicFee,
 			// set a much lower FeeCap than the actual maximum we are willing
 			// to pay in order to give ourselves headroom for bumping
 			// See: https://github.com/ethereum/go-ethereum/issues/24284
-			feeCap = calcFeeCap(b.latestBaseFee, b.config, tipCap)
+			feeCap = calcFeeCap(b.latestBaseFee, b.config, tipCap, maxGasPriceWei)
```

Comment:
Above this in line 229:

```		
		if b.config.EvmGasBumpThreshold() == 0 {
			// just use the max gas price if gas bumping is disabled
			feeCap = b.config.EvmMaxGasPriceWei()
```
Do we ever have gas bumping disabled? If so, would we want to use the more specific `maxGasPriceWei` there as well?

Reply:
I think @vreff 's comment is valid. There can potentially be scenarios where `EvmEIP1559DynamicFees` is enabled and `EvmGasBumpThreshold` is set to 0. In this case, we should honor `maxGasPriceWei`

Reply:
Yes you may wish to add it to `GetLegacyGas` also